packages/notebook/src/aviary/envs/notebook/env.py

```diff
@@ -350,10 +350,12 @@ async def close(self):
             # Have to do this since wildcard expansion doesn't work
             if self.use_tmp_work_dir:
                 await self._exec_cmd(["sh", "-c", "rm -r /workspace/*"])
-            await self.state.close()
-            await self.state._docker_client.close()
+            if hasattr(self, "state"):
```

Comment:
```suggestion
            if hasattr(self, "state"):  # State may not have been installed yet
```

Just explaining why we do this

Reply:
I guess we don't have typing protection here since it's like a runtime/stateful thing.

And the comment was just documenting why we have this conditional, as it's somewhat unexpected we can't do `self.state` at first glance